oxidation/client/src/components/SettingsModal.vue

```diff
@@ -0,0 +1,344 @@
+                  </ion-select-option>
+                </ion-select>
+              </ion-item>
+            </ion-list>
+          </div>
+          <!-- advanced -->
+          <div
+            v-else
+            class="settings-advanced"
+          >
+            <ion-list class="settings-list">
+              <!-- send error report -->
+              <settings-option
+                :title="$t('SettingsPage.enableTelemetry')"
+                :description="$t('SettingsPage.enableTelemetryDescription')"
+                v-model="config.enableTelemetry"
+              />
+              <!-- minimise in status bar -->
+              <settings-option
+                :title="$t('SettingsPage.minimizeToSystemTray')"
+                :description="$t('SettingsPage.minimizeToSystemTrayDescription')"
+                v-model="config.minimizeToTray"
+              />
+              <!-- display unsync files -->
+              <settings-option
+                :title="$t('SettingsPage.unsyncFiles')"
+                :description="$t('SettingsPage.unsyncFilesDescription')"
+                v-model="config.unsyncFiles"
+              />
+            </ion-list>
+          </div>
+        </div>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script setup lang = "ts" >
+import {
+  IonPage,
+  IonHeader,
+  IonToolbar,
+  IonTitle,
+  IonButtons,
+  IonButton,
+  IonContent,
+  IonList,
+  IonRadioGroup,
+  IonItem,
+  IonRadio,
+  IonText,
+  IonIcon,
+  IonSelect,
+  IonSelectOption,
+  modalController
+} from '@ionic/vue';
+
+import {
+  close,
+  cog,
+  options
+} from 'ionicons/icons';
+import { ref, inject, toRaw, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { onMounted, onUnmounted } from '@vue/runtime-core';
+import { toggleDarkMode } from '@/states/darkMode';
+import { Config, StorageManager } from '@/services/storageManager';
+import { storageManagerKey } from '@/main';
+import SettingsOption from './SettingsOption.vue';
+
+const { locale } = useI18n();
+const storageManager = inject(storageManagerKey)!;
+const config = ref<Config>(structuredClone(StorageManager.DEFAULT_CONFIG));
+const showTOS = ref('general');
+
+function closeModal(): Promise<boolean> {
+  return modalController.dismiss(null, 'cancel');
+}
+
+const configUnwatch = watch(config, async (_, oldConfig) => {
+  if (JSON.stringify(toRaw(oldConfig)) !== JSON.stringify(StorageManager.DEFAULT_CONFIG)) {
+    await storageManager.storeConfig(toRaw(config.value));
+  }
+  console.log(config.value);
+}, { deep: true });
+
+async function changeLang(selectedLang: string): Promise<void> {
+  config.value.locale = selectedLang;
+  locale.value = selectedLang;
+  await storageManager.storeConfig(toRaw(config.value));
+}
+
+async function changeTheme(selectedTheme: string): Promise<void> {
+  config.value.theme = selectedTheme;
+  toggleDarkMode(selectedTheme);
+  await storageManager.storeConfig(toRaw(config.value));
```

Comment:
Remove the call to storageManager here. The function `changeTheme` is only useful to force the changes of theme globally so that the whole application gets its new appearance. Since we also update `config.value.theme`, the watcher will be called and will do the `storeConfig` for us.